googlecloud-plugins/docs/BigQuery-batchsource.md

```diff
@@ -0,0 +1,69 @@
+# Google BigQuery Batch Source
+
+
+Description
+-----------
+Batch source to use Google BigQuery as a Source.
+[link to BigQuery Docs](https://cloud.google.com/bigquery/docs/)
+
+Use Case
+--------
+This source is used whenever you need to read from a Google BigQuery table or a BigQuery result.
+
+
+Properties
+----------
+**referenceName:** This will be used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**projectID:** The ID of the project in Google Cloud. (Macro-enabled)
+
+**jsonKeyFilePath:** The credential json key file path. (Macro-enabled)
+[How to get the keyfile](https://cloud.google.com/storage/docs/authentication#generating-a-private-key)
+
+**inputTableId:** The BigQuery table to read from, in the form [optional projectId]:[datasetId].[tableId], Example:
+publicdata:samples.shakespeare. Note that if the import query is specified, this table should be a empty table
+with the query result schema. User need to first create such a table.(Macro-enabled)
+
+**tempBucketPath:** The temporary Google Cloud Storage (GCS) directory to store the intermediate result.
+e.g. gs://bucketname/directoryname. The directory should not exist.
+shUser should delete this directory afterward manually to avoid extra google storage charge. (Macro-enabled)
+
+**importQuery:** The SELECT query to use to import data from the specified table. e.g.
+SELECT TOP(corpus, 10) as title, COUNT(*) as unique_words FROM [publicdata:samples.shakespeare].
+'publicdata' is the project name, smalples is the dataset name, shakespare is the table name.
+This is optional, if empty, just read the  inputTable configured. (Macro-enabled)
+
+**outputSchema:** Comma separated mapping of column names in the output schema to the data types; for example:
+'A:string,B:int'. (Macro-enabled)
+
+Example
+-------
+This example will process the import query on the [publicdata:samples.shakespeare], store the result in the inputTable,
+then download the inputTable to the tempBucketPath in Google Cloud Storage and finally reads the file in the temporary
```

Comment:
This description sounds like two temp files are being used, not just one. One for the results from the query, and then a second one for the download (?), and then the data gets read. Can this be done with one? In any case, I think some renaming is called for here, as calling something `inputTableId` but using it as a temp table is confusing.